lib/hardware_service.py

```diff
@@ -0,0 +1,12 @@
+class HardwareService(object):
```

Comment:
@vsemp as we have discussed,there is a lot of code already implemented for the hardware_service class and its concrete subclasses in the decouple-juniper branch. Can you branch off that, and then add your changes there, building on what has already been done? Then we can reuse the code we have already implemented and tested, and not have to completely redesign everything.